src/groupeddataframe/grouping.jl

```diff
@@ -190,42 +248,70 @@ end
 
 """
     combine(gd::GroupedDataFrame)
-    combine(f::Function, gd::GroupedDataFrame)
+    combine(f, gd::GroupedDataFrame)
 
 Transform a `GroupedDataFrame` into a `DataFrame`.
-If a function `f` is provided, it is called for each group in `gd` with a `SubDataFrame` view
-holding the corresponding rows, and the returned `DataFrame` then consists of the returned rows
-plus the grouping columns.
+
+If the first argument is a `cols => f` pair, `cols` must be a column name or index, or
+a vector or tuple thereof, and `f` must be a function. If `cols` is a single column index,
+`f` is passed a `SubArray` view into that column for each group; else, `f` is passed a
+named tuple holding `SubArray` views into these columns.
+
+If the first argument is a function, it is passed a `SubDataFrame` view for each group,
+and the returned `DataFrame` then consists of the returned rows plus the grouping columns.
+Note that this second form is much slower than the first one due to type instability.
 
 `f` can return a single value, a row or multiple rows. The type of the returned value
 determines the shape of the resulting data frame:
 - A single value gives a data frame with a single column and one row per group.
-- A named tuple or a `DataFrameRow` gives a data frame with one column for each field
-  and one row per group.
+- A named tuple of single values or a `DataFrameRow` gives a data frame with one column
+  for each field and one row per group.
 - A vector gives a data frame with a single column and as many rows
   for each group as the length of the returned vector for that group.
-- A data frame or a matrix gives a data frame with the same columns and as many rows
-  for each group as the rows returned for that group.
+- A data frame, a named tuple of vectors or a matrix gives a data frame
+  with the same columns and as many rows for each group as the rows returned for that group.
```

Comment:
In the cases where we generate column names we should somewhere in the documentation describe the rules for generating them.